bitnami/prometheus-operator/templates/_helpers.tpl

```diff
@@ -0,0 +1,377 @@
+{{- default .Chart.Name .Values.nameOverride | trunc 50 | trimSuffix "-" -}}
+{{- end -}}
+
+{{/* Name suffixed with operator */}}
+{{- define "prometheus-operator.operator.name" -}}
+{{- printf "%s-operator" (include "prometheus-operator.name" .) -}}
+{{- end }}
+
+{{/* Name suffixed with prometheus */}}
+{{- define "prometheus-operator.prometheus.name" -}}
+{{- printf "%s-prometheus" (include "prometheus-operator.name" .) -}}
+{{- end }}
+
+{{/* Name suffixed with alertmanager */}}
+{{- define "prometheus-operator.alertmanager.name" -}}
+{{- printf "%s-alertmanager" (include "prometheus-operator.name" .) -}}
+{{- end }}
+
+{{/*
+Create a default fully qualified app name.
+We truncate at 63 chars because some Kubernetes name fields are limited to this (by the DNS naming spec).
+*/}}
+{{- define "prometheus-operator.fullname" -}}
+{{- if .Values.fullnameOverride -}}
+{{- .Values.fullnameOverride | trunc 26 | trimSuffix "-" -}}
+{{- else -}}
+{{- $name := default .Chart.Name .Values.nameOverride -}}
+{{- if contains $name .Release.Name -}}
+{{- .Release.Name | trunc 26 | trimSuffix "-" -}}
+{{- else -}}
+{{- printf "%s-%s" .Release.Name $name | trunc 26 | trimSuffix "-" -}}
+{{- end -}}
+{{- end -}}
+{{- end -}}
+
+{{/* Fullname suffixed with operator */}}
+{{- define "prometheus-operator.operator.fullname" -}}
+{{- printf "%s-operator" (include "prometheus-operator.fullname" .) -}}
+{{- end }}
+
+{{/* Fullname suffixed with prometheus */}}
+{{- define "prometheus-operator.prometheus.fullname" -}}
+{{- printf "%s-prometheus" (include "prometheus-operator.fullname" .) -}}
+{{- end }}
+
+{{/* Fullname suffixed with alertmanager */}}
+{{- define "prometheus-operator.alertmanager.fullname" -}}
+{{- printf "%s-alertmanager" (include "prometheus-operator.fullname" .) -}}
+{{- end }}
+
+{{- define "prometheus-operator.chart" -}}
+{{- printf "%s-%s" .Chart.Name .Chart.Version | replace "+" "_" | trunc 63 | trimSuffix "-" -}}
+{{- end -}}
+
+{{/*
+Common Labels
+*/}}
+{{- define "prometheus-operator.labels" -}}
+helm.sh/chart: {{ include "prometheus-operator.chart" . }}
+app.kubernetes.io/instance: {{ .Release.Name }}
+app.kubernetes.io/managed-by: {{ .Release.Service }}
+{{- if .Values.global.labels }}
+{{ toYaml .Values.global.labels }}
+{{- end }}
+{{- end -}}
+
+{{/*
+Labels for operator
+*/}}
+{{- define "prometheus-operator.operator.labels" -}}
+app.kubernetes.io/name: {{ include "prometheus-operator.operator.name" . }}
+{{ include "prometheus-operator.labels" . }}
+{{- end -}}
+
+{{/*
+Labels for prometheus
+*/}}
+{{- define "prometheus-operator.prometheus.labels" -}}
+app.kubernetes.io/name: {{ include "prometheus-operator.prometheus.name" . }}
+{{ include "prometheus-operator.labels" . }}
+{{- end -}}
+
+{{/*
+Labels for alertmanager
+*/}}
+{{- define "prometheus-operator.alertmanager.labels" -}}
+app.kubernetes.io/name: {{ include "prometheus-operator.alertmanager.name" . }}
+{{ include "prometheus-operator.labels" . }}
+{{- end -}}
```

Comment:
I'm not very sure about the labels you chose... Shouldn't you use sth like this?
```
{{/*
Common Labels
*/}}
{{- define "prometheus-operator.labels" -}}
app.kubernetes.io/name: {{ include "prometheus-operator.name" . }}
helm.sh/chart: {{ include "prometheus-operator.chart" . }}
app.kubernetes.io/instance: {{ .Release.Name }}
app.kubernetes.io/managed-by: {{ .Release.Service }}
{{- if .Values.global.labels }}
{{ toYaml .Values.global.labels }}
{{- end }}
{{- end -}}

{{/*
Labels for operator
*/}}
{{- define "prometheus-operator.operator.labels" -}}
{{ include "prometheus-operator.labels" . }}
app.kubernetes.io/component: operator
{{- end -}}

{{/*
Labels for prometheus
*/}}
{{- define "prometheus-operator.prometheus.labels" -}}
{{ include "prometheus-operator.labels" . }}
app.kubernetes.io/component: prometheus
{{- end -}}

{{/*
Labels for alertmanager
*/}}
{{- define "prometheus-operator.alertmanager.labels" -}}
{{ include "prometheus-operator.labels" . }}
app.kubernetes.io/component: alertmanager
{{- end -}}
```

Reply:
point taken